practical/DeepClustering/kmeans_Cimsir.py

```diff
@@ -0,0 +1,109 @@
+from typing import Iterator
+
+import numpy as np
+
+import torch
+import torch.nn as nn
+import torch.utils.data.dataloader
+import torch.optim
+
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+
+from sklearn.datasets import make_blobs
+from scipy.spatial import distance
+from torch.nn import Parameter
+
+
+class KMeans:
+    """
+    Standard k-Means Algorithm
+    """
+
+    def __init__(self, n_cluster, distance=distance.euclidean):
```

Comment:
Don't add the distance function of choice via the __init__ class method. This is bad practice, as this could result in future errors when another user wants to use another and/or custom distance function. Instead, offer a choice, such as 'euclidean' and offer a general purpose function such as distance.cdist.
Another advantage: cdist will calculate all distances between to d-Dimensional arrays, which should save compute time with larger datasets.